samples/samples_utils.h

```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+
+typedef struct top_sym_t {
```

Comment:
Likewise, should we add a source comment saying this struct should not be used and adding a TODO to remove it?